### PR TITLE
Fix input name inconsistency in womtool responses [BA-6021]

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/graph/ScatterElementToGraphNode.scala [hunk 1]
```diff
@@ -150,10 +150,11 @@ object ScatterElementToGraphNode {
...
-        case i: RequiredInputDefinition => i.name -> RequiredGraphInputNode(WomIdentifier(i.name), i.womType, i.name, Callable.InputDefinition.IdentityValueMapper)
-        case i: OptionalInputDefinition => i.name -> OptionalGraphInputNode(WomIdentifier(i.name), i.womType, i.name, Callable.InputDefinition.IdentityValueMapper)
-        case i: OverridableInputDefinitionWithDefault => i.name -> OptionalGraphInputNodeWithDefault(WomIdentifier(i.name), i.womType, i.default, i.name, Callable.InputDefinition.IdentityValueMapper)
+        case i: RequiredInputDefinition => i.name -> RequiredGraphInputNode(inputNodeIdentifier(i), i.womType, a.workflowName + "." + i.name, Callable.InputDefinition.IdentityValueMapper)
+        case i: OptionalInputDefinition => i.name -> OptionalGraphInputNode(inputNodeIdentifier(i), i.womType, a.workflowName + "." + i.name, Callable.InputDefinition.IdentityValueMapper)
+        case i: OverridableInputDefinitionWithDefault => i.name -> OptionalGraphInputNodeWithDefault(inputNodeIdentifier(i), i.womType, i.default, a.workflowName + "." + i.name, Callable.InputDefinition.IdentityValueMapper)
```
Comment (cjllanwarne) [inline wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/graph/ScatterElementToGraphNode.scala]: Note: This was the fix to the originally reported bug. Almost everything else in the PR was tidying up of side-effects of this change.

File: wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/graph/CallElementToGraphNode.scala [hunk 2]
```diff
@@ -139,8 +145,16 @@ object CallElementToGraphNode {
 
         // No input mapping, add an optional input using the default expression
         case withDefault@OverridableInputDefinitionWithDefault(n, womType, expression, _, _) =>
-          val identifier = WomIdentifier(s"${a.workflowName}.$callName.${n.value}")
-          withGraphInputNode(withDefault, OptionalGraphInputNodeWithDefault(identifier, womType, expression, identifier.fullyQualifiedName.value))
+          val identifier = WomIdentifier(
+            localName = s"$callName.${n.value}",
+            fullyQualifiedName = s"${a.workflowName}.$callName.${n.value}"
+          )
```
Comment (cjllanwarne) [inline wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/graph/CallElementToGraphNode.scala]: NB: this change to how identifiers are constructed for inputs was the fix of the original bug. Almost everything else was caused by rippling side-effects of this fix

File: services/src/test/scala/cromwell/services/womtool/impl/WomtoolServiceInCromwellActorSpec.scala [hunk 3]
```diff
@@ -313,10 +316,13 @@ class WomtoolServiceInCromwellActorSpec extends ServicesSpec("Womtool") {
...
-      result <- (womtoolActor ? request).mapTo[DescribeResult]
-      _ = result shouldBe expectedResponse
-    } yield ()).futureValue
+    val result = Await.result((womtoolActor ? request).mapTo[DescribeResult], 10.seconds)
+    result shouldBe expectedResponse
+
+//    (for {
+//      result <- (womtoolActor ? request).mapTo[DescribeResult]
+//      _ = result shouldBe expectedResponse
+//    } yield ()).futureValue
```
Comment (kshakir) [inline services/src/test/scala/cromwell/services/womtool/impl/WomtoolServiceInCromwellActorSpec.scala]: Remove?

File: wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/graph/CallElementToGraphNode.scala [hunk 4]
```diff
@@ -52,6 +52,15 @@ object CallElementToGraphNode {
         case None => s"Cannot resolve a callable with name ${a.node.callableReference}".invalidNel
       }
 
+    def supplyableInput(definition: Callable.InputDefinition): Boolean = {
+        !definition.isInstanceOf[FixedInputDefinitionWithDefault] &&
+        !definition.name.contains(".") // NB: Remove this check when sub-workflows allow pass-through task inputs
```
Comment (kshakir) [inline wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/graph/CallElementToGraphNode.scala]: ToL: Does this deserve/have a ticket, or is it fine if it's never fixed/tracked/seen-again?

Comment (cjllanwarne) [reply wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/graph/CallElementToGraphNode.scala]: This is more of a "if WDL 2.0 removes this requirement (🤞) then this line goes away"

**Pull request merged**